addons/website/static/src/js/editor/widget_link.js

```diff
@@ -66,7 +67,14 @@ weWidgets.LinkDialog.include({
 
         function always() {
             $anchorsLoading.addClass('d-none');
-            $selectMenu.prop("selectedIndex", -1);
+            const anchor = `#${$selectMenu.data("anchor-for").split('#')[1]}`;
```

Comment:
Why not `urlInputValue` instead of `$selectMenu.data("anchor-for")` ? At this point `$selectMenu.data("anchor-for")` could not be even set to the right thing (if you go through the `else` above).

Reply:
> Why not `urlInputValue` instead of `$selectMenu.data("anchor-for")` ? At this point `$selectMenu.data("anchor-for")` could not be even set to the right thing (if you go through the `else` above).

Actually, due to the time that `loadAnchors` takes, `$selectMenu.data("anchor-for", urlInputValue);` is executed before it completes. That is why it works..

> if you go through the else above

If it goes through the "else", both have the same value, so it doesn't matter in this case.

I prefered to use  `$selectMenu.data("anchor-for")` instead of `urlInputValue` because `urlInputValue` is not accessible in the `always` function and I want to reduce lines changes. In fact, it might be best to move `$selectMenu.prop("selectedIndex", anchorIndex);` into the `if` (before `loadAnchors`) as we don't need it in the `else` anyway. Let me know what you think...

Reply:
> because urlInputValue is not accessible in the always function and I want to reduce lines changes

Why would not it be accessible in the `always` function? :thinking: 

> In fact, it might be best to move $selectMenu.prop("selectedIndex", anchorIndex); into the if (before loadAnchors) as we don't need it in the else anyway.

I think I prefer it in the `always` ("once xxx is done, update the UI" -> no matter if xxx should impact the UI or not).

Reply:
> I think I prefer it in the `always` ("once xxx is done, update the UI" -> no matter if xxx should impact the UI or not).

Oops, I made a mistake when copying and pasting that code. What I actually meant was `$selectMenu.data("anchor-for", urlInputValue);`.

Reply:
> Why would not it be accessible in the `always` function? thinking

Indeed I was wrong. I made the change.